container/kubernetes/onboarding/deviceonboarding.py

```diff
@@ -0,0 +1,396 @@
+    # Clone into temporary dir
+    repo = Repo.clone_from(git_url, tempdir)
+    if git_tag:
+        repo.head.reset(git_tag, index=True, working_tree=True)
+    # Copy desired file from temporary dir
+    # if config_type DEPLOYMENT ENUM
+    if config_type == CONFIG.DEPLOYMENT:
+        filelist = os.listdir(os.path.join(tempdir, git_dir))
+        for filename in filelist:
+            shutil.move(os.path.join(tempdir, git_dir, filename),
+                        os.path.join(appname_dir, filename))
+    elif config_type == CONFIG.GATEWAY:
+        dirlist = get_all_subdir_paths(os.path.join(tempdir, git_dir))
+        if "gateway" in dirlist[0]:
+            dirlist.reverse()
+        # We need to copy the dataservice application yaml
+        # as generic one to distingush between gatewayapp
+        shutil.move(os.path.join(tempdir, git_dir,
+                                 dirlist[0], 'application.yaml'),
+                    os.path.join(appname_dir,
+                                 'fusiondataservice_application.yaml'))
+        shutil.move(os.path.join(tempdir, git_dir, dirlist[1],
+                                 'application.yaml'),
+                    os.path.join(appname_dir, 'application.yaml'))
+    # Remove temporary dir
+    shutil.rmtree(tempdir)
+
+
+def get_current_directory():
+    """Return current working directory."""
+    prog = __file__
+    return os.path.dirname(os.path.abspath(prog))
+
+
+def signal_activationcode():
+    """Check activation code expiration status
+        and re-generate it, if expired.
+    """
+    cmd = "find activation-code -mmin -60 | wc -l"
+    process = subprocess.Popen(
+        cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    check_status = process.communicate()[0].decode('utf-8').strip()
+    # Cosmetic extraction of data from subprocess command
+    # If "activation-code" file doesn't exists
+    if '\n' in check_status:
+        check_status = check_status.splitlines()[1]
+
+    if int(check_status) == 0:
+        # Check the Status of Activation code sub-process
+        # and exit the process incase of failure.
+        ret = subprocess.call(['sh', 'environment_setup.sh'])
+        if ret != 0:
+            sys.exit()
+
+
+def parsing_device_configdata(deploydata):
+    """ Returns multidict list from parsed onboarding
+        devices's machine and depployment config's.
+    """
+    # Parsing gateway*.yaml and creating list of deployment's.
+    # Input sample List of Deploymen and Each Deployment is List of
+    # Deployment and Machine Config Elements:
+    # [{
+    # 'ApplicationName': 'gasversorung', 'NodeName': 'shailesh-nuc',
+    # 'ToolVersion': 1,
+    # 'DeploymentConfig':
+    #  {  'URL': 'https://github.com/shailesh83/Gateway-Deployment',
+    #    'Directory': 'deploy-gas',
+    #      'GitTag': '650541f301347d469297b9cfcddf39b3dce52370'
+    #  },
+    # 'MachineConfig':
+    #  { 'URL': 'https://github.com/mattmikulina/
```

Comment:
please remove the urls. Even in comments, not acceptable

Reply:
Add something generic like {MACHINE_CONFIG_URL} would be my suggestion.

Reply:
yes, comments i missed, will finish it..

Reply:
done.